lib/evaluate/variable.cc

```diff
@@ -27,6 +27,12 @@ using namespace Fortran::parser::literals;
 
 namespace Fortran::evaluate {
 
+int GetSymbolRank(const Symbol *symbol) { return symbol->Rank(); }
```

Comment:
Random thought: would it help at all to have a type SymbolRef that encapsulates the whole pointer-to-immutable-object syntactic fluff?

Reply:
Should the rank concept be encapsulated to enforce correctness? As a bare int, one could theoretically get a rank of -45288, for instance.

Reply:
I would prefer a reference to a pointer here, but I wouldn't want to see a typedef for either.  I can tell what `const Symbol &` means without having to look at any other definition.

Reply:
I'll change it to a reference.